pkg/mount/mount.go

```diff
@@ -78,17 +80,25 @@ func RecursiveUnmount(target string) error {
 		}
 		logrus.Debugf("Trying to unmount %s", m.Mountpoint)
 		err = Unmount(m.Mountpoint)
-		if err != nil && i == len(mounts)-1 {
-			if mounted, err := Mounted(m.Mountpoint); err != nil || mounted {
-				return err
+		if err != nil {
+			// If the error is EINVAL either this whole package is wrong (invalid flags passed to unmount(2)) or this is
+			// not a mountpoint (which is ok in this case).
+			// Meanwhile calling `Mounted()` is very expensive.
+			if err == syscall.EINVAL {
+				continue
+			}
+			if i == len(mounts)-1 {
+				if mounted, e := Mounted(m.Mountpoint); e != nil || mounted {
+					return err
+				}
+				continue
 			}
-			// Ignore errors for submounts and continue trying to unmount others
-			// The final unmount should fail if there ane any submounts remaining
-		} else if err != nil {
-			logrus.Errorf("Failed to unmount %s: %v", m.Mountpoint, err)
-		} else if err == nil {
-			logrus.Debugf("Unmounted %s", m.Mountpoint)
+			// This is some submount, we can ignore this error for now, the final unmount will fail if this is a real problem
+			logrus.WithError(err).Debugf("Failed to unmount %s", m.Mountpoint)
```

Comment:
Perhaps it make sense to raise level to warning (as this message will be a good hint about why the final umount failed), as well as saying this is a submount. Something like
```go
logrus.WithError(err).Warnf("Failed ti umount submount %s", m.Mountpoint)
```